### PR TITLE
feat: Implement additional simple properties for topic and subscription

### PR DESCRIPTION
**Pull request by zucchinidev**
[#187726204](https://www.pivotaltracker.com/story/show/187726204)

### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: google-pubsub.yml [hunk 1]
```diff
@@ -34,6 +34,64 @@ provision:
...
+      Retain acknowledged messages.
+      If true, then messages are not expunged from the subscription's backlog, even if they are acknowledged,
+      until they fall out of the messageRetentionDuration window.
+    default: false
+  - field_name: subscription_expiration_policy
+    type: string
+    details: |
+      Duration after which the subscription expires if no messages are received (e.g., "604800s" for 7 days)
+      If expirationPolicy is not set, a default policy with ttl of 31 days will be used.
+      If it is set with "", the resource never expires. The minimum allowed value is 1 day
```
Comment (FelisiaM) [inline google-pubsub.yml:67]: These two sentences seem to be contradicting. 

> If expirationPolicy is not set, a default policy with ttl of 31 days will be used. 
> If it is set with "", the resource never expires.

So when the customer does not explicitly set the `subscription_expiration_policy` we will default to `""` which means we don't pass anything to tofu so it defaults to 31 days.

If the customer passes explicitly  `""` we don't pass anything to tofu so it defaults to 31 days and not to never expire.

File: terraform/pubsub/provision/variables.tf [hunk 2]
```diff
@@ -7,4 +7,14 @@ variable "credentials" {
   type      = string
   sensitive = true
 }
-variable "project" { type = string }
\ No newline at end of file
+variable "project" { type = string }
+
+variable "topic_message_retention_duration" { type = string }
```
Comment (FelisiaM) [inline terraform/pubsub/provision/variables.tf:12]: nitp: I would try to keep the same ordering as in the yml definition (so move labels/credentials/project) to the bottom

**Review by FelisiaM — CHANGES_REQUESTED** [review]

File: terraform-tests/pubsub_test.go [hunk 3]
```diff
@@ -20,13 +20,22 @@ var _ = Describe("PubSub", Label("pubsub-terraform"), Ordered, func() {
...
+			"credentials":                      googleCredentials,
+			"project":                          googleProject,
+			"labels":                           map[string]string{"label1": "value1"},
+			"topic_name":                       "test-topic-name",
+			"subscription_name":                "",
+			"ack_deadline":                     10,
+			"push_endpoint":                    "",
+			"topic_message_retention_duration": "",
+			"topic_kms_key_name":               "",
+			"subscription_message_retention_duration":   "",
```
Comment (FelisiaM) [inline terraform-tests/pubsub_test.go:32]: The formatting here seems to be a bit off

Comment (zucchinidev) [reply terraform-tests/pubsub_test.go:32]: I know, but when formatting, the fmt tool does not change anything :thinking:

File: google-pubsub.yml (hunk 1 above)
Comment (zucchinidev) [reply google-pubsub.yml:67]: Great catch!! Thank you very much @FelisiaM . I am fixing it now

Comment (zucchinidev) [reply google-pubsub.yml]: Fixed! Notice that once the value is set, we can no longer unset it!

**Review by FelisiaM — APPROVED** [review]

**Pull request merged**